### ISSUE TITLE
An mapper specified for a Enum is not being considered by anotation processor

### ISSUE THREAD

**Issue by cvgaviao** [vert-x3/vertx-codegen#295]
### Version

4.0.0-SNAPSHOT

### Context

Some of my data objects uses a [Enum](https://github.com/cvgaviao/vertx-codegen-failing/blob/master/src/main/java/org/example/RunMode.java) that has a custom constructor. Because of this I've created a  [converter](https://github.com/cvgaviao/vertx-codegen-failing/blob/master/src/main/java/org/example/utils/MyMappers.java) for it using static methods.
The converter is being mapped using `json-mappers.properties`.
The DataObject's converter class is being generated, but its fromJson() method is using the generic pattern for the enum:
`obj.setMode(org.example.RunMode.valueOf((String)member.getValue()));`

### Do you have a reproducer?

https://github.com/cvgaviao/vertx-codegen-failing

**Comment by vietj** [comment]
@cvgaviao we need to review all these PR's

**Comment by vietj** [comment]
as far as I understand you want to map an enum to a string but use your own converter right ?

**Comment by vietj** [comment]
and also, this is a use case for service proxies right ?

**Comment by cvgaviao** [comment]
> as far as I understand you want to map an enum to a string but use your own converter right ?

Yep, the main necessity was to initially map a string property in a JSON object into a Enum, but those Enums doesn't have a defaut constructor. So, the need for a custom converter to do it.

**Comment by vietj** [comment]
it is still not clear because string are mapped to enum using `valueOf`, not sure what you mean by constructor, can you elaborate ?

**Comment by cvgaviao** [comment]
> it is still not clear because string are mapped to enum using `valueOf`, not sure what you mean by constructor, can you elaborate ?

You can only use valueOf if, and only if, your JSON string property is EXACTLY (including case) equal to the Enum constant. Take a look in the Enum [example](https://github.com/vert-x3/vertx-codegen/pull/325/files#diff-6d29d6699be8cc816fbda973ede584fa0426ba6aeda9c738ca957dd368a1303c) that I've added in the PR:

**Comment by vietj** [comment]
that was my question @cvgaviao no worries.
